### ISSUE TITLE
kubectl describe doesn't show annotations 

### ISSUE THREAD

**Issue by Miyurz** [kubernetes/kubernetes#44914]
**What keywords did you search in Kubernetes issues before filing this one?** 
kubectl annotation 

---

**BUG REPORT:** 
 I did a `kubectl describe aws-cli` to check the annotations but it didn't show any. 
My pod definition, aws-cli.yml is as follows:
```
---
apiVersion: v1
kind: Pod
metadata:
  name: aws-cli
  labels:
    name: aws-cli
  annotations:
    iam.amazonaws.com/role: arn:aws:iam::487267539104:role/Engineer
spec:
  containers:
  - image: fstab/aws-cli
    command:
      - "/home/aws/aws/env/bin/aws"
      - "s3"
      - "ls"
      - "kops-production-euwest-1"
    name: aws-cli
```
I researched through a couple of bugs including https://github.com/kubernetes/kubernetes/issues/23380 to look for a probable fix.

**Kubernetes version** (use `kubectl version`):
```
$ kubectl version
Client Version: version.Info{Major:"1", Minor:"5", GitVersion:"v1.5.2", GitCommit:"08e099554f3c31f6e6f07b448ab3ed78d0520507", GitTreeState:"clean", BuildDate:"2017-01-12T04:57:25Z", GoVersion:"go1.7.4", Compiler:"gc", Platform:"linux/amd64"}
Server Version: version.Info{Major:"1", Minor:"5", GitVersion:"v1.5.2", GitCommit:"08e099554f3c31f6e6f07b448ab3ed78d0520507", GitTreeState:"clean", BuildDate:"2017-01-12T04:52:34Z", GoVersion:"go1.7.4", Compiler:"gc", Platform:"linux/amd64"}
```

**Environment**:
- **Cloud provider or hardware configuration**:
AWS
- **OS** (e.g. from /etc/os-release):
Ubuntu 16.04
- **Kernel** (e.g. `uname -a`):
Linux vagrant 4.4.0-51-generic #72-Ubuntu SMP Thu Nov 24 18:29:54 UTC 2016 x86_64 x86_64 x86_64 GNU/Linux
- **Install tools**:
kops, helm

**What happened**:
On adding an annotation to aws.cli.yml , the pod went into containerCreating state. so I ran a `kubectl describe pod aws-cli` to view the meta data and annotation but surprisingly it was missing. I researched through a few bugs addressing a similar issue and looked like the fix was merged way back.

**What you expected to happen**:
annotations should have shown up in the describe pod. If not, is there any other way to view them ?

**How to reproduce it** (as minimally and precisely as possible):
Create the pod and just  spin it in the cluster. run a `kubectl describe <pod>` and look out for annotations

**Comment by xiangpengzhao** [comment]
You can run `kubectl get pod aws-cli -o json` to view almost all the info of your pod `aws-cli`

**Comment by xiangpengzhao** [comment]
v1.5.2 doesn't display annotations yet, see code:
https://github.com/kubernetes/kubernetes/blob/v1.5.2/pkg/kubectl/describe.go#L472

#35094 implemented it and it's in v1.6.

**Comment by smarterclayton** [comment]
Fixed in the latest release.
